frame/democracy/src/lib.rs

```diff
@@ -1677,19 +1818,102 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Current era is ending; we should finish up any proposals.
-	fn begin_block(now: T::BlockNumber) -> DispatchResult {
+	///
+	/// If a referendum is launched or maturing take full block weight.
+	fn begin_block(now: T::BlockNumber) -> Result<Weight, DispatchError> {
+		// TODO TODO: do idle weight using benchmark on_initialize_no_launch_no_maturing
+		let mut weight = 0;
 		// pick out another public referendum if it's time.
 		if (now % T::LaunchPeriod::get()).is_zero() {
```

Comment:
doesn't this mean that we require at least one DB read and return the weight accordingly?

Reply:
Yes the `let mut weight = 0;` should be rewritten with the value of the new benchmark implementation in this PR `on_initliaze_no_launch_no_maturing`